libopenage/pathfinding/cost_field.cpp

```diff
@@ -33,29 +34,42 @@ cost_t CostField::get_cost(size_t idx) const {
 	return this->cells.at(idx);
 }
 
-void CostField::set_cost(const coord::tile_delta &pos, cost_t cost) {
+void CostField::set_cost(const coord::tile_delta &pos, cost_t cost, time::time_t &changed) {
 	this->cells[pos.ne + pos.se * this->size] = cost;
+	this->changed = changed;
 }
 
-void CostField::set_cost(size_t x, size_t y, cost_t cost) {
+void CostField::set_cost(size_t x, size_t y, cost_t cost, time::time_t &changed) {
 	this->cells[x + y * this->size] = cost;
+	this->changed = changed;
 }
 
 void CostField::set_cost(size_t idx, cost_t cost) {
 	this->cells[idx] = cost;
+	this->changed = time::TIME_ZERO;
+}
```

Comment:
See map.cpp, lines 50-52:
```
				auto sector = grid->get_sector(chunk_idx);
				auto cost_field = sector->get_cost_field();
				cost_field->set_cost(tile_idx, path_cost.second);
```
Not sure how else to initialize cost fields with a time stamp without passing some reference to the internal clock into map.cpp. So what I did was leave that method without a time parameter, setting the time to default to the start of the runtime. What route should be taken here to better resolve this?

Reply:
If it's for initialization, you can just pass `time::TIME_ZERO` from the map. I wouldn't worry about how the map does updates yet, since that is a different cattle of fish. However, you can assume for now that the update time will always be passed down the line (from the event that updates the map) and the clock is not necessary. You only need the clock in very few cases where it's necessary to know the current game time.